Tools/Scripts/webkitpy/style/checkers/basexcconfig.py

```diff
@@ -0,0 +1,95 @@
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from webkitpy.common.memoized import memoized
+
+
+"""Enforces rules for Base.xcconfig files."""
+
+
+class BaseXcconfigChecker(object):
+    categories = set([
+        'basexcconfig/missing-wk_default-prefix',
+        'basexcconfig/missing-inherited',
+        'basexcconfig/overrides-common-base',
+    ])
+
+    default_vars = {
+        'GCC_OPTIMIZATION_LEVEL': 1,
+    }
+
+    def __init__(self, file_path, handle_style_error):
+        self._file_path = file_path
+        self._handle_style_error = handle_style_error
+
+    @memoized
+    def read_common_base_xcconfig_variables(self):
+        inherited_vars = {}
+        override_vars = {}
+
+        for line in open('Configurations/CommonBase.xcconfig'):
+            # Find lines containing assignments.
+            lhs, operator, rhs = map(str.strip, line.partition('='))
+
+            # Skip non-assignment lines, comments, and WK_ variables which are allowed to be overridden.
+            if operator != '=' or lhs.startswith('//') or lhs.startswith('WK_'):
+                continue
+
+            # Discard any setting condition.
+            name = lhs.partition('[')[0]
+
+            if '$(inherited)' in rhs:
+                inherited_vars[name] = 1
+            elif not self.default_vars.get(name):
+                override_vars[name] = 1
+
+        return inherited_vars, override_vars
+
+    def check(self, lines):
+        (inherited_vars, override_vars) = self.read_common_base_xcconfig_variables()
+
+        for line_number, line in enumerate(lines, start=1):
+            # Find lines containing assignments.
+            lhs, operator, rhs = map(str.strip, line.partition('='))
+
+            # Skip non-assignment lines, comments, and WK_ variables which are allowed to be overridden.
+            if operator != '=' or lhs.startswith('//') or lhs.startswith('WK_'):
+                continue
+
+            # Discard any setting condition.
+            name = lhs.partition('[')[0]
+
+            if self.default_vars.get(name):
+                self._handle_style_error(
+                    line_number, 'basexcconfig/missing-wk_default-prefix', 5,
```

Comment:
Regarding `line_number`, just want to make sure that this number is supposed to be relative to wherever `lines` occurs in the source file.

Reply:
I believe `line_number` is supposed to match the line number for that line in the source file.

Should I not have the `start=1` parameter?

Reply:
IIRC, the checker may only give you lines that changed, so `start=1` would not always be accurate.

Reply:
If I add this to `Configurations/CommonBase.xcconfig`:

```
CLANG_CXX_LANGUAGE_STANDARD = c++2a;
```

And then run `check-webkit-style`, the line numbers are correct.  I think it's okay to leave it now, and do a follow-up fix if it needs an update.